pynamodb/attributes.py

```diff
@@ -835,14 +827,14 @@ def _update_attribute_paths(self, path_segment):
             if isinstance(local_attr, MapAttribute):
                 local_attr._update_attribute_paths(path_segment)
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> 'Comparison':  # type: ignore[override]
         if self._is_attribute_container():
-            return AttributeContainer.__eq__(self, other)
+            return self is other  # type: ignore
         return Attribute.__eq__(self, other)
 
-    def __ne__(self, other):
+    def __ne__(self, other: Any) -> 'Comparison':  # type: ignore[override]
```

Comment:
Do you think it's worth adding mypy tests for this?

Reply:
Yup! Added.